Bottom: 47364b9634df8a832343929afd47f0cedb3f836f
Top:    6d41e3d0032c07f307905c76acc1a14281fcb536
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-04 23:45:14 +0000

More hack to avoid GL calls when we're out of context


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index b8b2e2b..b28f6da 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -506,6 +506,12 @@ ghid_set_color (hidGC gc, const char *name)
 
   gc->colorname = (char *) name;
 
+  if (!check_gl_drawing_ok_hack)
+    {
+      current_color = NULL;
+      return;
+    }
+
   if (gport->colormap == NULL)
     gport->colormap = gtk_widget_get_colormap (gport->top_window);
   if (strcmp (name, "erase") == 0)
